internal/literals/literals.go

```diff
@@ -87,55 +87,117 @@ func Obfuscate(file *ast.File, info *types.Info, fset *token.FileSet, ignoreObj
 
 			switch y := info.TypeOf(x.Type).(type) {
 			case *types.Array:
-				if y.Elem() != byteType {
-					return true
-				}
-				if y.Len() > maxSizeBytes {
-					return true
-				}
-
-				data := make([]byte, y.Len())
-
-				for i, el := range x.Elts {
-					lit, ok := el.(*ast.BasicLit)
-					if !ok {
+				if y.Elem() == byteType {
+					if y.Len() > maxSizeBytes {
 						return true
 					}
 
-					value, err := strconv.Atoi(lit.Value)
-					if err != nil {
-						return true
+					data := make([]byte, y.Len())
+
+					for i, el := range x.Elts {
+						lit, ok := el.(*ast.BasicLit)
+						if !ok {
+							return true
+						}
+
+						// if the element is a rune literal, parse it as such
+						var value byte
+						if lit.Value[0] == '\'' {
+							val, _, _, err := strconv.UnquoteChar(lit.Value, 0)
+							if err != nil {
+								panic(fmt.Sprintf("cannot parse rune: %v", err))
+							}
+							value = byte(val)
```

Comment:
you should document why it's safe to convert from rune to byte here, as in general that can cause overflows

Reply:
I dont think its safe